### PR TITLE
Support views and foreign tables in yii\pgsql\Schema.findTableNames

### PR DESCRIPTION
**Pull request by nineinchnick**
References #7833.

### CODE REVIEW

**Comment by samdark** [comment]
Is it possible to add some tests for it?

**Comment by nineinchnick** [comment]
Sure, it's a good idea since I was wondering if there will be a difference between an ordinary and materialized view.

**Comment by nineinchnick** [comment]
I've added test but actually ran only sqlite and pgsql.

I also found and fixed one bug in mssql tests introduced by dd15857, too much copy and paste and `innodb` was left in mssql schema. But that just means nobody executed those tests since Feb16.

File: framework/db/pgsql/Schema.php [hunk 1]
```diff
@@ -177,8 +177,10 @@ protected function findTableNames($schema = '')
             $schema = $this->defaultSchema;
         }
         $sql = <<<EOD
-SELECT table_name, table_schema FROM information_schema.tables
-WHERE table_schema=:schema AND table_type='BASE TABLE'
+SELECT c.relname AS table_name
```
Comment (samdark) [inline framework/db/pgsql/Schema.php]: `table_schema` isn't selected anymore. Are you sure it's OK?

Comment (nineinchnick) [inline framework/db/pgsql/Schema.php]: It wasn't used.

Comment (samdark) [inline framework/db/pgsql/Schema.php]: OK.

**Comment by nineinchnick** [comment]
@samdark this one is also ready for merge

**Comment by samdark** [comment]
Except changelog :) Also it seems there are conflicts with master now.

**Comment by nineinchnick** [comment]
@samdark done, but somehow I messed up the `skip CI` part.

**Comment by samdark** [comment]
Merged https://github.com/yiisoft/yii2/commit/29304a185abe6a4627eac8e062e333928a1bac85 Thanks!

**Comment by cebe** [comment]
@nineinchnick the syntax is a bit unnatural: `[ci skip]`

**Pull request closed without merging**